platform/os_windows.go

```diff
@@ -189,6 +206,170 @@ func SetSdnRemoteArpMacAddress() error {
 	return nil
 }
 
+func HasMellanoxAdapater() bool {
+	adapterName, err := getMellanoxAdapterName()
+	if err != nil {
+		log.Errorf("Error while getting mellanox adapter name: %v", err)
+		return false
+	} else if adapterName == "" {
+		log.Printf("No Mellanox adapter found")
+		return false
+	}
+	log.Printf("Name of Mellanox adapter : %v", adapterName)
+	return true
+}
+
+// Regularly monitors the Mellanox PriorityVLANGTag registry value and sets it to desired value if needed
+func MonitorAndSetMellanoxRegKeyPriorityVLANTag() {
+	for {
+		adapterName, err := getMellanoxAdapterName()
+		if err == nil && adapterName != "" {
+			err := SetMellanoxPriorityVLANTag(adapterName)
+			if err != nil {
+				log.Errorf("error while monitoring and setting Mellanox Reg Key value: %v", err)
+			}
+			time.Sleep(mellanoxPriorityVLANTagMonitorInterval)
+		} else {
+			log.Printf("getMellanoxAdapterName returned err: %v and adapterName: %s", err, adapterName)
+		}
+	}
+}
+
+func getMellanoxAdapterName() (string, error) {
+	//get mellanox adapter name
+	cmd := fmt.Sprintf("Get-NetAdapter | Where-Object { $_.InterfaceDescription -like \"%s\" } | Select-Object -ExpandProperty Name", mellanoxSearchString)
+	adapterName, err := ExecutePowershellCommand(cmd)
+	if err != nil {
+		return "", fmt.Errorf("error while executing powershell command to get net adapter list: %w", err)
+	}
+	if adapterName == "" {
+		log.Printf("no network adapter found with %s in description", mellanoxSearchString)
+	}
+	return adapterName, nil
+}
+
+// Set Mellanox adapter's PriorityVLANTag value to 3 if adapter exists
+// reg key value for PriorityVLANTag = 3  --> Packet priority and VLAN enabled
+// for more details goto https://docs.nvidia.com/networking/display/winof2v230/Configuring+the+Driver+Registry+Keys#ConfiguringtheDriverRegistryKeys-GeneralRegistryKeysGeneralRegistryKeys
+func SetMellanoxPriorityVLANTag(adapterName string) error {
+	//Find if adapter has property PriorityVLANTag (version 4 or up) or not (version 3)
+	cmd := fmt.Sprintf("Get-NetAdapterAdvancedProperty | Where-Object { $_.RegistryKeyword -like \"%s\" -and $_.Name -eq \"%s\" } | Select-Object -ExpandProperty Name", priorityVLANTagIdentifier, adapterName)
+	adapterNameWithVLANTag, err := ExecutePowershellCommand(cmd)
+	if err != nil {
+		return fmt.Errorf("error while executing powershell command to get VLAN Tag advance property of %s: %w", adapterName, err)
+	}
+
+	if adapterNameWithVLANTag != "" {
```

Comment:
if you flip the condition and return early, most of this function will be unindented, which is more readable.

Reply:
this if condition branches off on Mellanox adapter version so we cannot return early even if we flip the condition. In both if and else, we attempt to set Mellanox version through different PowerShell commands.